controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -301,10 +306,141 @@ func (r *ReplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	err = r.processLastActionResult(ctx, localRG, remoteClient, log)
+	if err != nil {
+		r.EventRecorder.Eventf(localRG, eventTypeWarning, eventReasonUpdated,
+			"failed to process the last action %s", localRG.Status.LastAction.Condition)
+	}
+
 	log.V(common.InfoLevel).Info("RG has already been synced to the remote cluster")
 	return ctrl.Result{}, nil
 }
 
+func (r *ReplicationGroupReconciler) processLastActionResult(ctx context.Context, group *repv1.DellCSIReplicationGroup, remoteClient connection.RemoteClusterClient, log logr.Logger) error {
+	if group.Status.LastAction.ErrorMessage != "" {
+		return fmt.Errorf("error in processing the last action")
+	}
+
+	if len(group.Status.Conditions) == 0 || group.Status.LastAction.Time == nil {
+		log.V(common.InfoLevel).Info("No action to process")
+		return nil
+	}
+
+	if strings.Contains(group.Status.LastAction.Condition, "CREATE_SNAPSHOT") {
+		return r.processSnapshotEvent(ctx, group, remoteClient, log)
+	}
+
+	return nil
+}
+
+func (r *ReplicationGroupReconciler) processSnapshotEvent(ctx context.Context, group *repv1.DellCSIReplicationGroup, remoteClient connection.RemoteClusterClient, log logr.Logger) error {
+	lastAction := group.Status.LastAction
+	lastTime := lastAction.Time
+	currTime := time.Now()
+	diffTime := currTime.Sub(lastTime.Time).Seconds()
+
+	if diffTime > 30 {
```

Comment:
How often does Reconcile RG happens? If it runs within 30 sec, will the snapshot objects be created again on target k8s cluster?

Reply:
Reconcile happens every minute unless the RG actions are updated.

There is no good way of determining whether to process except when the action was completed. 30 seconds seemed like a good threshold so that we don't duplicate snapshot creation if they are executed too quickly.

We could keep track of the last time a create snapshot action was executed and only create the objects when they don't match. 

Thoughts?

Reply:
Added a new annotation to inform the controller that an action should be processed. This can be extended for the use of other actions in the future if needed.